core/shared/src/main/scala/interface.scala

```diff
@@ -155,6 +155,26 @@ abstract class CaseClass[Typeclass[_], Type] private[magnolia] (
   final def construct[Return](makeParam: Param[Typeclass, Type] => Return): Type =
     rawConstruct(parameters map makeParam)
 
+  /**
+   * Like construct but allows each parameter to fail with an error.
+   *
+   * @see construct
+   */
+  final def econstruct[E <: AnyRef, Return](makeParam: Param[Typeclass, Type] => Either[E, Return]): Either[E, Type] = {
+    // poor man's scalaz.Traverse
+    try {
+      val bits = parameters.map { p =>
+        makeParam(p) match {
+          case Left(e) => throw EarlyExit(e)
```

Comment:
`return Left(err.asInstanceOf[E])`, and remove the `try`/`catch`?

Reply:
doesn't work, because the nearest method seems to be the map closure (I tried it)